onnxruntime/test/contrib_ops/math/matmul_sparse_test.cc

```diff
@@ -0,0 +1,387 @@
+//  tester.AddSparseCsrOutput("B", {2, 3}, float_values, {0, 2, 5}, {0, 2, 3}, CheckParams());
+//
+//  // Output span
+//  tester.AddSparseCooOutput("A", {2, 3}, gsl::make_span(float_values), {0, 4}, CheckParams());
+//  tester.AddSparseCsrOutput("B", {2, 3}, gsl::make_span(float_values), {0, 2, 5}, {0, 2, 3}, CheckParams());
+//}
+
+template<typename T>
+void ConvertToCsr(gsl::span<const T> input_span,
+  const std::vector<int64_t>& dims,
+  std::vector<T>& values_out,
+  std::vector<int64_t>& inner_indicies_out,
+  std::vector<int64_t>& outer_indicies_out) {
+
+  ASSERT_EQ(dims.size(), 2U);
+  const auto rows = dims[0];
+  const auto cols = dims[1];
+  const auto dense_size = rows * cols;
+  ASSERT_EQ(input_span.size(), static_cast<size_t>(dense_size));
+
+  const int64_t nnz = std::count_if(input_span.cbegin(), input_span.cend(),
+                                    [](T v) { return v != T(0); });
+
+  std::vector<T> values;
+  std::vector<int64_t> inner;
+  std::vector<int64_t> outer;
+
+  if (nnz > 0) {
+    values.reserve(nnz);
+    inner.reserve(nnz);
+    outer.reserve(rows + 1);
+
+    outer.push_back(0);
+    int64_t col = 0;
+    for (int64_t i = 0; i < dense_size; ++i, ++col) {
+      if (col >= cols) {
+        outer.push_back(static_cast<int64_t>(values.size()));
+        col = 0;
+      }
+      if (input_span[i] != T(0)) {
+        values.push_back(input_span[i]);
+        inner.push_back(col);
+      }
+    }
+    // Final entry
+    outer.push_back(static_cast<int64_t>(values.size()));
+  }
+
+  values_out = std::move(values);
+  inner_indicies_out = std::move(inner);
+  outer_indicies_out = std::move(outer);
+}
+
+// Converts with 2-D indices
+template<typename T>
+void ConvertToCoo(gsl::span<const T> input_span,
+  const std::vector<int64_t>& dims,
+  std::vector<T>& values_out,
+  std::vector<int64_t>& indices_out) {
+
+  ASSERT_EQ(dims.size(), 2U);
+  const auto rows = dims[0];
+  const auto cols = dims[1];
+  const auto dense_size = rows * cols;
+  ASSERT_EQ(input_span.size(), static_cast<size_t>(dense_size));
+
+  const int64_t nnz = std::count_if(input_span.cbegin(), input_span.cend(),
+                                    [](T v) { return v != T(0); });
+
+  std::vector<T> values;
+  std::vector<int64_t> indices;
+  if (nnz > 0) {
+    values.reserve(nnz);
+    indices.reserve(2 * nnz);
+
+    int64_t row = 0;
+    int64_t col = 0;
+    for (int64_t i = 0; i < dense_size; ++i, ++col) {
+      if (col >= cols) {
+        col = 0;
+        ++row;
+      }
+      if (input_span[i] != T(0)) {
+        values.push_back(input_span[i]);
+        indices.push_back(row);
+        indices.push_back(col);
+      }
+    }
+  }
+  values_out = std::move(values);
+  indices_out = std::move(indices);
+}
+
+/// This test must be disabled x86 build because of an apparent Eigen bug
+/// However, things work on x64 builds
+/// It manifests itself as Status Message: std::bad_alloc or Status Message: bad allocation
```

Comment:
So if a test breaks on x86 would running a model also potentially break (or worse silently return invalid results)?

Reply:
Is it intentional to disable for `__wasm__` as well? The comment didn't mention if that platform was affected.

Reply:
Will update the comment. Nodejs (__wasm__) is built with 32-bit and it was first manifested there.